src/Repository/Geo/ZoneRepository.php

```diff
@@ -51,4 +52,24 @@ public function zoneableAsZone(ZoneableInterface $zoneable): Zone
 
         return $zone;
     }
+
+    /**
+     * @param Zone[] $zones
+     */
+    public function createSelectByParentsQueryBuilder(array $zones): QueryBuilder
+    {
+        $qb = $this->createQueryBuilder('zone');
+
+        return $qb
+            ->leftJoin('zone.children', 'children')
+            ->andWhere(
+                $qb->expr()->orX(
+                    $qb->expr()->in('zone.id', ':zones'),
+                    $qb->expr()->in('children.id', ':zones'),
```

Comment:
We need to check parents in the zone, not children.
For exemple, if a department is passed to the function, we need to get all boroughs, cities, canton etc and department itself.

Reply:
done in 239bb7948be8369f930b3e7a0c58b9252b4ada1d